### PR TITLE
Synthesis of encoding circuits.

### PR DESCRIPTION
**Pull request by jmbr**

### CODE REVIEW

File: src/quil/stdgates.quil [hunk 1]
```diff
@@ -14,6 +14,10 @@ DEFGATE Y:
     0, -i
     i, 0
 
+DEFGATE YY:
```
Comment (stylewarning) [inline src/quil/stdgates.quil]: v dangerous of you to add to the QUIL standard.....

Comment (jmbr) [reply src/quil/stdgates.quil]: ...hence the "do not merge" label.

File: src/quilec/code.lisp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+(defun make-code (generators)
+  "Helper function to instantiate CODE classes."
+  (flet ((get-generators (generators)
+           "Pre-process the sequence GENERATORS, which is expected to contain valid inputs to PAULI-FROM-SYMBOLS and return the corresponding array of PAULI objects."
+           (let (number-of-physical-qubits
+                 effective-generators)
+             (dolist (term
+                      (coerce generators 'list)
+                      (values number-of-physical-qubits
+                              (coerce (nreverse effective-generators) 'generators)))
```
Comment (jlapeyre) [inline src/quilec/code.lisp]: Trailing whitespace on this line.

**Comment by notmgsk** [comment]
I think something has gone awry. This branch appears to be including changes that were already in master. For example, the diff includes the Magicl NG diff.

**Comment by jmbr** [comment]
> I think something has gone awry. This branch appears to be including changes that were already in master. For example, the diff includes the Magicl NG diff. And this PR now touches 48 files.

I'm aware of it. Thanks.

**Pull request closed without merging**